Algorithm.CSharp/SmaCrossUniverseSelectionAlgorithm.cs

```diff
@@ -0,0 +1,85 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using QuantConnect.Data.UniverseSelection;
+using QuantConnect.Indicators;
+using System.Collections.Concurrent;
+using System.Linq;
+
+namespace QuantConnect.Algorithm.CSharp
+{
+    /// <summary>
+    /// Provides an example where WarmUpIndicator method is used to warm up indicators
+    /// after their security is added and before (Universe Selection scenario)
+    /// </summary>
+    public class SmaCrossUniverseSelectionAlgorithm : QCAlgorithm
+    {
+        private const int _count = 10;
+        private const decimal _tolerance = 0.01m;
+        private const decimal _targetPercent = 1m / _count;
+
+        public override void Initialize()
+        {
+            UniverseSettings.Leverage = 2.0m;
+            UniverseSettings.Resolution = Resolution.Daily;
+
+            SetStartDate(2018, 01, 01);
+            SetEndDate(2019, 01, 01);
+            SetCash(1000000);
+
+            IsWarmUpIndicatorEnabled = true;
+
+            var symbol = AddEquity("SPY", Resolution.Hour).Symbol;
+            var ema = EMA(symbol, 10);
+            WarmUpIndicator(symbol, ema);
```

Comment:
Please add a similar test case for an indicator on 
1) custom data e.g. Quandl
2) quotebar data,
3) ticks.